Cannot compile Objective-C++ file if Ably headers included

### Which version of the Ably SDK are you using? 
`1.1.14`

### On which platform does the issue happen? 
`iOS` at compilation time

### Are you using Carthage? 
No.

### Are you using Cocoapods? 
Yes, version `1.8.4`.

### Which version of Xcode are you using? 
`11.2.1`

### What did you do?
Imported the Ably client library in to an Objective-C++ source file (`.mm` extension) using:

```
#import <Ably/Ably.h>
```

### What did you expect to happen?
Xcode to build successfully.

### What happened instead?
Build failed with compiler error.

```

Showing Recent Issues
CompileC /Users/quintin/Library/Developer/Xcode/DerivedData/TestObjectiveCPlusPlus-axyypleyxtiwgndvlladmbaergpn/Build/Intermediates.noindex/TestObjectiveCPlusPlus.build/Debug-iphonesimulator/TestObjectiveCPlusPlus.build/Objects-normal/x86_64/TestCPlusPlus.o /Users/quintin/code/experiments/ably/cocoa/TestObjectiveCPlusPlus/TestObjectiveCPlusPlus/TestCPlusPlus.mm normal x86_64 objective-c++ com.apple.compilers.llvm.clang.1_0.compiler (in target 'TestObjectiveCPlusPlus' from project 'TestObjectiveCPlusPlus')
    cd /Users/quintin/code/experiments/ably/cocoa/TestObjectiveCPlusPlus
    export LANG=en_US.US-ASCII
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -x objective-c++ -target x86_64-apple-ios13.2-simulator -fmessage-length=0 -fdiagnostics-show-note-include-stack -fmacro-backtrace-limit=0 -std=gnu++14 -stdlib=libc++ -fobjc-arc -fobjc-weak -fmodules -gmodules -fmodules-cache-path=/Users/quintin/Library/Developer/Xcode/DerivedData/ModuleCache.noindex -fmodules-prune-interval=86400 -fmodules-prune-after=345600 -fbuild-session-file=/Users/quintin/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/Session.modulevalidation -fmodules-validate-once-per-build-session -Wnon-modular-include-in-framework-module -Werror=non-modular-include-in-framework-module -Wno-trigraphs -fpascal-strings -O0 -fno-common -Wno-missing-field-initializers -Wno-missing-prototypes -Werror=return-type -Wdocumentation -Wunreachable-code -Wno-implicit-atomic-properties -Werror=deprecated-objc-isa-usage -Wno-objc-interface-ivars -Werror=objc-root-class -Wno-arc-repeated-use-of-weak -Wimplicit-retain-self -Wno-non-virtual-dtor -Wno-overloaded-virtual -Wno-exit-time-destructors -Wduplicate-method-match -Wno-missing-braces -Wparentheses -Wswitch -Wunused-function -Wno-unused-label -Wno-unused-parameter -Wunused-variable -Wunused-value -Wempty-body -Wuninitialized -Wconditional-uninitialized -Wno-unknown-pragmas -Wno-shadow -Wno-four-char-constants -Wno-conversion -Wconstant-conversion -Wint-conversion -Wbool-conversion -Wenum-conversion -Wno-float-conversion -Wnon-literal-null-conversion -Wobjc-literal-conversion -Wshorten-64-to-32 -Wno-newline-eof -Wno-selector -Wno-strict-selector-match -Wundeclared-selector -Wdeprecated-implementations -Wno-c++11-extensions -DDEBUG=1 -DCOCOAPODS=1 -DOBJC_OLD_DISPATCH_PROTOTYPES=0 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator13.2.sdk -fasm-blocks -fstrict-aliasing -Wprotocol -Wdeprecated-declarations -Winvalid-offsetof -g -fvisibility-inlines-hidden -Wno-sign-conversion -Winfinite-recursion -Wmove -Wcomma -Wblock-capture-autoreleasing -Wstrict-prototypes -Wrange-loop-analysis -Wno-semicolon-before-method-body -Wunguarded-availability -fobjc-abi-version=2 -fobjc-legacy-dispatch -index-store-path /Users/quintin/Library/Developer/Xcode/DerivedData/TestObjectiveCPlusPlus-axyypleyxtiwgndvlladmbaergpn/Index/DataStore -iquote /Users/quintin/Library/Developer/Xcode/DerivedData/TestObjectiveCPlusPlus-axyypleyxtiwgndvlladmbaergpn/Build/Intermediates.noindex/TestObjectiveCPlusPlus.build/Debug-iphonesimulator/TestObjectiveCPlusPlus.build/TestObjectiveCPlusPlus-generated-files.hmap -I/Users/quintin/Library/Developer/Xcode/DerivedData/TestObjectiveCPlusPlus-axyypleyxtiwgndvlladmbaergpn/Build/Intermediates.noindex/TestObjectiveCPlusPlus.build/Debug-iphonesimulator/TestObjectiveCPlusPlus.build/TestObjectiveCPlusPlus-own-target-headers.hmap -I/Users/quintin/Library/Developer/Xcode/DerivedData/TestObjectiveCPlusPlus-axyypleyxtiwgndvlladmbaergpn/Build/Intermediates.noindex/TestObjectiveCPlusPlus.build/Debug-iphonesimulator/TestObjectiveCPlusPlus.build/TestObjectiveCPlusPlus-all-non-framework-target-headers.hmap -ivfsoverlay /Users/quintin/Library/Developer/Xcode/DerivedData/TestObjectiveCPlusPlus-axyypleyxtiwgndvlladmbaergpn/Build/Intermediates.noindex/TestObjectiveCPlusPlus.build/Debug-iphonesimulator/TestObjectiveCPlusPlus.build/all-product-headers.yaml -iquote /Users/quintin/Library/Developer/Xcode/DerivedData/TestObjectiveCPlusPlus-axyypleyxtiwgndvlladmbaergpn/Build/Intermediates.noindex/TestObjectiveCPlusPlus.build/Debug-iphonesimulator/TestObjectiveCPlusPlus.build/TestObjectiveCPlusPlus-project-headers.hmap -I/Users/quintin/Library/Developer/Xcode/DerivedData/TestObjectiveCPlusPlus-axyypleyxtiwgndvlladmbaergpn/Build/Products/Debug-iphonesimulator/include -I/Users/quintin/Library/Developer/Xcode/DerivedData/TestObjectiveCPlusPlus-axyypleyxtiwgndvlladmbaergpn/Build/Products/Debug-iphonesimulator/Ably/Ably.framework/Headers -I/Users/quintin/Library/Developer/Xcode/DerivedData/TestObjectiveCPlusPlus-axyypleyxtiwgndvlladmbaergpn/Build/Products/Debug-iphonesimulator/KSCrashAblyFork/KSCrashAblyFork.framework/Headers -I/Users/quintin/Library/Developer/Xcode/DerivedData/TestObjectiveCPlusPlus-axyypleyxtiwgndvlladmbaergpn/Build/Products/Debug-iphonesimulator/MsgPackAblyFork/MsgPackAblyFork.framework/Headers -I/Users/quintin/Library/Developer/Xcode/DerivedData/TestObjectiveCPlusPlus-axyypleyxtiwgndvlladmbaergpn/Build/Products/Debug-iphonesimulator/SAMKeychain/SAMKeychain.framework/Headers -I/Users/quintin/Library/Developer/Xcode/DerivedData/TestObjectiveCPlusPlus-axyypleyxtiwgndvlladmbaergpn/Build/Products/Debug-iphonesimulator/SocketRocketAblyFork/SocketRocketAblyFork.framework/Headers -I/Users/quintin/Library/Developer/Xcode/DerivedData/TestObjectiveCPlusPlus-axyypleyxtiwgndvlladmbaergpn/Build/Products/Debug-iphonesimulator/ULID/ULID.framework/Headers -I/Users/quintin/Library/Developer/Xcode/DerivedData/TestObjectiveCPlusPlus-axyypleyxtiwgndvlladmbaergpn/Build/Intermediates.noindex/TestObjectiveCPlusPlus.build/Debug-iphonesimulator/TestObjectiveCPlusPlus.build/DerivedSources-normal/x86_64 -I/Users/quintin/Library/Developer/Xcode/DerivedData/TestObjectiveCPlusPlus-axyypleyxtiwgndvlladmbaergpn/Build/Intermediates.noindex/TestObjectiveCPlusPlus.build/Debug-iphonesimulator/TestObjectiveCPlusPlus.build/DerivedSources/x86_64 -I/Users/quintin/Library/Developer/Xcode/DerivedData/TestObjectiveCPlusPlus-axyypleyxtiwgndvlladmbaergpn/Build/Intermediates.noindex/TestObjectiveCPlusPlus.build/Debug-iphonesimulator/TestObjectiveCPlusPlus.build/DerivedSources -F/Users/quintin/Library/Developer/Xcode/DerivedData/TestObjectiveCPlusPlus-axyypleyxtiwgndvlladmbaergpn/Build/Products/Debug-iphonesimulator -F/Users/quintin/Library/Developer/Xcode/DerivedData/TestObjectiveCPlusPlus-axyypleyxtiwgndvlladmbaergpn/Build/Products/Debug-iphonesimulator/Ably -F/Users/quintin/Library/Developer/Xcode/DerivedData/TestObjectiveCPlusPlus-axyypleyxtiwgndvlladmbaergpn/Build/Products/Debug-iphonesimulator/KSCrashAblyFork -F/Users/quintin/Library/Developer/Xcode/DerivedData/TestObjectiveCPlusPlus-axyypleyxtiwgndvlladmbaergpn/Build/Products/Debug-iphonesimulator/MsgPackAblyFork -F/Users/quintin/Library/Developer/Xcode/DerivedData/TestObjectiveCPlusPlus-axyypleyxtiwgndvlladmbaergpn/Build/Products/Debug-iphonesimulator/SAMKeychain -F/Users/quintin/Library/Developer/Xcode/DerivedData/TestObjectiveCPlusPlus-axyypleyxtiwgndvlladmbaergpn/Build/Products/Debug-iphonesimulator/SocketRocketAblyFork -F/Users/quintin/Library/Developer/Xcode/DerivedData/TestObjectiveCPlusPlus-axyypleyxtiwgndvlladmbaergpn/Build/Products/Debug-iphonesimulator/ULID -MMD -MT dependencies -MF /Users/quintin/Library/Developer/Xcode/DerivedData/TestObjectiveCPlusPlus-axyypleyxtiwgndvlladmbaergpn/Build/Intermediates.noindex/TestObjectiveCPlusPlus.build/Debug-iphonesimulator/TestObjectiveCPlusPlus.build/Objects-normal/x86_64/TestCPlusPlus.d --serialize-diagnostics /Users/quintin/Library/Developer/Xcode/DerivedData/TestObjectiveCPlusPlus-axyypleyxtiwgndvlladmbaergpn/Build/Intermediates.noindex/TestObjectiveCPlusPlus.build/Debug-iphonesimulator/TestObjectiveCPlusPlus.build/Objects-normal/x86_64/TestCPlusPlus.dia -c /Users/quintin/code/experiments/ably/cocoa/TestObjectiveCPlusPlus/TestObjectiveCPlusPlus/TestCPlusPlus.mm -o /Users/quintin/Library/Developer/Xcode/DerivedData/TestObjectiveCPlusPlus-axyypleyxtiwgndvlladmbaergpn/Build/Intermediates.noindex/TestObjectiveCPlusPlus.build/Debug-iphonesimulator/TestObjectiveCPlusPlus.build/Objects-normal/x86_64/TestCPlusPlus.o

In file included from /Users/quintin/code/experiments/ably/cocoa/TestObjectiveCPlusPlus/TestObjectiveCPlusPlus/TestCPlusPlus.mm:10:
In file included from /Users/quintin/Library/Developer/Xcode/DerivedData/TestObjectiveCPlusPlus-axyypleyxtiwgndvlladmbaergpn/Build/Products/Debug-iphonesimulator/Ably/Ably.framework/Headers/Ably.h:24:
/Users/quintin/Library/Developer/Xcode/DerivedData/TestObjectiveCPlusPlus-axyypleyxtiwgndvlladmbaergpn/Build/Products/Debug-iphonesimulator/Ably/Ably.framework/Headers/ARTHttp.h:40:35: error: expected identifier; 'class' is a keyword in Objective-C++
+ (void)setURLSessionClass:(Class)class;
                                  ^
1 error generated.
```

--- a/Source/ARTHttp.h
+++ b/Source/ARTHttp.h
@@ -37,7 +37,7 @@ NS_ASSUME_NONNULL_BEGIN
 
 @interface ARTHttp : NSObject<ARTHTTPExecutor>
 
-+ (void)setURLSessionClass:(Class)class;
++ (void)setURLSessionClass:(Class)urlSessionClass;
 
 - (instancetype)init UNAVAILABLE_ATTRIBUTE;
 - (instancetype)init:(dispatch_queue_t)queue logger:(ARTLog *)logger;

--- a/Source/ARTHttp.m
+++ b/Source/ARTHttp.m
@@ -25,8 +25,8 @@ Class configuredUrlSessionClass = nil;
     _Nullable dispatch_queue_t _queue;
 }
 
-+ (void)setURLSessionClass:(Class)class {
-    configuredUrlSessionClass = class;
++ (void)setURLSessionClass:(const Class)urlSessionClass {
+    configuredUrlSessionClass = urlSessionClass;
 }
 
 - (instancetype)init:(dispatch_queue_t)queue logger:(ARTLog *)logger {

--- a/Source/ARTWebSocketTransport+Private.h
+++ b/Source/ARTWebSocketTransport+Private.h
@@ -17,7 +17,7 @@ NS_ASSUME_NONNULL_BEGIN
 
 @interface ARTWebSocketTransport () <ARTWebSocketDelegate>
 
-+ (void)setWebSocketClass:(Class)class;
++ (void)setWebSocketClass:(Class)webSocketClass;
 
 // From RestClient
 @property (readwrite, strong, nonatomic) id<ARTEncoder> encoder;

--- a/Source/ARTWebSocketTransport.m
+++ b/Source/ARTWebSocketTransport.m
@@ -56,8 +56,8 @@ Class configuredWebsocketClass = nil;
 @synthesize delegate = _delegate;
 @synthesize stateEmitter = _stateEmitter;
 
-+ (void)setWebSocketClass:(Class)class {
-    configuredWebsocketClass = class;
++ (void)setWebSocketClass:(const Class)webSocketClass {
+    configuredWebsocketClass = webSocketClass;
 }
 
 - (instancetype)initWithRest:(ARTRestInternal *)rest options:(ARTClientOptions *)options resumeKey:(NSString *)resumeKey connectionSerial:(NSNumber *)connectionSerial {

